planner/core/testdata/plan_suite_in.json

```diff
@@ -1094,5 +1094,24 @@
       "SELECT ta.NAME FROM ta WHERE EXISTS (select /*+ no_decorrelate() */ 1 from tb where ta.code = tb.code and tb.NAME LIKE 'chad9%') AND (select /*+ no_decorrelate() */ max(id) from tc where ta.name=tc.name and tc.name like 'chad99%') > 100 and (select /*+ no_decorrelate() */ max(id) from td where ta.id=td.id and td.name like 'chad999%') > 100"
 
     ]
+  },
+  {
+    "name": "TestCountStarForTikv",
+    "cases": [
+      "select count(*) from t",
+      "select count(*) from t where a=1",
+      "select a, count(*) from t group by a -- shouldn't be rewritten"
+    ]
+  },
+  {
+    "name": "TestCountStarForTiFlash",
+    "cases": [
+      "select count(*) from t",
+      "select count(*) from t where a=1",
+      "select count(*) from t_pick_row_id",
+      "select count(distinct 1) from t -- shouldn't be rewritten",
+      "select count(1), count(a), count(b) from t -- count(1) to count(a)",
+      "select a, count(*) from t group by a -- shouldn't be rewritten"
```

Comment:
Can we have test cases where the aggregation is a child of a Join, or the count(*) aggregation is inside of a subquery in a Project/Filter node? Given that the rule is recursive down processing, I doesn't any test cases cover that code path.

Reply:
done